### PR TITLE
CVS-51577 ovms does not validate correcty dag with model used as demultiplexer node

### PR DESCRIPTION
**Pull request by Ficol**

### CODE REVIEW

File: src/pipelinedefinition.cpp [hunk 1]
```diff
@@ -617,8 +618,7 @@ class NodeValidator {
                 return result;
             }
 
-            if (
-                (dependantNodeInfo.kind == NodeKind::DL || dependantNodeInfo.kind == NodeKind::CUSTOM) &&
+            if (dependantNodeInfo.kind != NodeKind::ENTRY &&
```
Comment (dkalinowski) [inline src/pipelinedefinition.cpp:621]: Why this change? We allow checking connection metadata correctness for EXIT node, which has no defined input metadata.

File: src/pipelinedefinition.cpp [hunk 2]
```diff
@@ -469,16 +469,17 @@ class NodeValidator {
...
-        if (dependencyNodeInfo.demultiplyCount) {
-            auto result = influenceShapeWithDemultiplexer(tensorOutputShape, dependencyNodeInfo);
-            if (!result.ok()) {
-                return result;
-            }
+        auto result = influenceShapeWithDemultiplexer(tensorOutputShape, dependencyNodeInfo);
+        if (!result.ok()) {
+            return result;
+        }
+        if (dependantNodeInfo.kind == NodeKind::EXIT) {
```
Comment (dkalinowski) [inline src/pipelinedefinition.cpp:478]: This is kind of hack to force demultiplexer demultiplyCount check in `influenceShapeWithDemultiplexer`.
Maybe extract this check out of `influenceShapeWithDemultiplexer` into separate method and call outside of `checkConnectionMetadataCorrectness`?

**Review by atobiszei — COMMENTED** [review]
Could we test this with mocked modelinstance which shape could support demultiplexing?

**Review by dkalinowski — APPROVED** [review]

**Review by atobiszei — APPROVED** [review]
